### PR TITLE
feat: supports `withdrawTreasury` to an address

### PR DESCRIPTION
**Pull request by robertu7**

### CODE REVIEW

File: src/TheSpace/HarbergerMarket.sol [hunk 1]
```diff
@@ -185,13 +185,17 @@ contract HarbergerMarket is ERC721Enumerable, IHarbergerMarket, Multicall, ACLMa
...
+
+    function _getPrice(uint256 tokenId_) internal view returns (uint256 price) {
+        return tokenRecord[tokenId_].price;
     }
 
     /// @inheritdoc IHarbergerMarket
     function setPrice(uint256 tokenId_, uint256 price_) public {
         if (!_isApprovedOrOwner(msg.sender, tokenId_)) revert Unauthorized();
-        if (price_ == getPrice(tokenId_)) return;
+        if (price_ == _getPrice(tokenId_)) return;
```
Comment (robertu7) [inline src/TheSpace/HarbergerMarket.sol:198]: use `_getPrice` to avoid repetitive checks on `_exists`

File: src/TheSpace/HarbergerMarket.sol [hunk 2]
```diff
@@ -213,7 +216,7 @@ contract HarbergerMarket is ERC721Enumerable, IHarbergerMarket, Multicall, ACLMa
             if (owner == msg.sender) return;
 
             // check price
-            uint256 askPrice = getPrice(tokenId_);
+            uint256 askPrice = _getPrice(tokenId_);
```
Comment (robertu7) [inline src/TheSpace/HarbergerMarket.sol:219]: same reason above

File: src/TheSpace/HarbergerMarket.sol [hunk 3]
```diff
@@ -247,9 +249,8 @@ contract HarbergerMarket is ERC721Enumerable, IHarbergerMarket, Multicall, ACLMa
             // equal to bidding from address 0 with price 0
             emit Bid(tokenId_, address(0), msg.sender, 0);
 
-            // initialize tax record and price
-            tokenRecord[tokenId_].lastTaxCollection = block.number;
```
Comment (robertu7) [inline src/TheSpace/HarbergerMarket.sol:251]: L245 has already update `tokenRecord[tokenId_].lastTaxCollection`

File: src/TheSpace/HarbergerMarket.sol [hunk 4]
```diff
@@ -366,10 +368,18 @@ contract HarbergerMarket is ERC721Enumerable, IHarbergerMarket, Multicall, ACLMa
      * @dev Internel function to set price for a token.
      */
     function _setPrice(uint256 tokenId_, uint256 price_) internal {
+        _setPrice(tokenId_, price_, ownerOf(tokenId_));
+    }
+
+    function _setPrice(
```
Comment (robertu7) [inline src/TheSpace/HarbergerMarket.sol:374]: avoid re-read `_ownerOf` on `_setPrice`

File: src/TheSpace/HarbergerMarket.sol [hunk 5]
```diff
@@ -272,14 +276,15 @@ contract HarbergerMarket is ERC721Enumerable, IHarbergerMarket, Multicall, ACLMa
...
+
         if (tax > 0) {
             // calculate collectable amount
             address taxpayer = ownerOf(tokenId_);
             uint256 allowance = currency.allowance(taxpayer, address(this));
             uint256 balance = currency.balanceOf(taxpayer);
             uint256 available = allowance < balance ? allowance : balance;
 
-            if (available > tax) {
+            if (available >= tax) {
```
Comment (robertu7) [inline src/TheSpace/HarbergerMarket.sol:287]: `>` => `>=`

**Review by guoliu — APPROVED** [review]
🪴

**Pull request merged**